pkg/velaql/providers/query/handler.go

```diff
@@ -255,6 +256,55 @@ func (h *provider) GeneratorServiceEndpoints(wfctx wfContext.Context, v *value.V
 			for _, ing := range ingress {
 				serviceEndpoints = append(serviceEndpoints, generatorFromIngress(ing, cluster)...)
 			}
+		case "SeldonDeployment":
+			var vs istio.VirtualService
+			if err := findResource(&vs, resource.Name, resource.Namespace, resource.Cluster); err != nil {
+				klog.Error(err, fmt.Sprintf("find v1alpha3 VirtualService %s/%s from cluster %s failure", resource.Name, resource.Namespace, resource.Cluster))
+				continue
+			}
+			if vs.Spec.Http == nil || vs.Spec.Http[0].Match == nil {
+				klog.Error(fmt.Sprintf("find VirtualService %s/%s from cluster %s empty prefix", resource.Name, resource.Namespace, resource.Cluster))
+				continue
+			}
+			// get istio service from label here since the seldon's gateway is predefined
+			var istioService corev1.ServiceList
+			labels := &v1.LabelSelector{
+				MatchLabels: map[string]string{
+					"istio": "ingressgateway",
+				},
+			}
+			selector, err := v1.LabelSelectorAsSelector(labels)
+			if err != nil {
+				return err
+			}
+			if err := h.cli.List(ctx, &istioService, &client.ListOptions{
+				LabelSelector: selector,
+			}); err != nil || len(istioService.Items) == 0 {
+				klog.Error(err, fmt.Sprintf("find istio service from cluster %s failure", resource.Cluster))
+				continue
+			}
+			service := istioService.Items[0]
+			if service.Status.LoadBalancer.Ingress == nil {
+				klog.Error(fmt.Sprintf("find istio service from cluster %s empty ingress", resource.Cluster))
+				continue
+			}
+			serviceEndpoints = append(serviceEndpoints, querytypes.ServiceEndpoint{
+				Endpoint: querytypes.Endpoint{
+					Host:     service.Status.LoadBalancer.Ingress[0].IP,
```

Comment:
Array overflow risk?   Line 256 has the same problem

Reply:
Checked in line 287 and line 265